include/boost/geometry/algorithms/detail/disjoint/linear_areal.hpp

```diff
@@ -141,28 +141,26 @@ struct disjoint_segment_areal
 template <typename Segment, typename Polygon>
 class disjoint_segment_areal<Segment, Polygon, polygon_tag>
 {
-private:
+    
     template <typename InteriorRings, typename Strategy>
     static inline
     bool check_interior_rings(InteriorRings const& interior_rings,
                               Segment const& segment,
                               Strategy const& strategy)
     {
-        typedef typename boost::range_value<InteriorRings>::type ring_type;
+        using ring_type = typename boost::range_value<InteriorRings>::type;
 
-        typedef unary_disjoint_geometry_to_query_geometry
+        using unary_predicate_type = unary_disjoint_geometry_to_query_geometry
             <
                 Segment,
                 Strategy,
                 disjoint_range_segment_or_box<ring_type, Segment>
-            > unary_predicate_type;
-                
-        return check_iterator_range
-            <
-                unary_predicate_type
-            >::apply(boost::begin(interior_rings),
-                     boost::end(interior_rings),
-                     unary_predicate_type(segment, strategy));
+            >;
+
+        return boost::end(interior_rings) == 
+            std::find_if(boost::begin(interior_rings),
+                         boost::end(interior_rings), 
+                         unary_predicate_type(segment, strategy));     
```

Comment:
Find if a segment intersects any interior ring. If not found and also for empty range return `true`. Ok.

----

`end` is created 2 times and makes the formatting harder. What do you think about replacing it with the code below?
```
auto const end = boost::end(interior_rings);
return std::find_if(boost::begin(interior_rings), end, 
                    unary_predicate_type(segment, strategy)) == end;
```